core/src/main/java/org/apache/druid/java/util/common/parsers/ObjectFlatteners.java

```diff
@@ -32,13 +32,8 @@
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-public class ObjectFlatteners
+public abstract class ObjectFlatteners
```

Comment:
This seems like an unneeded change, I guess it achieves the same thing, but I think it's a bit more confusing because I at least expect a class to be abstract because it's going to be implemented somewhere, where as previously it was obviously just trying to be a static class

Reply:
Ah, I see now, it has just one static method. I was confused since it felt like an abstract class to me. Do you want to revert this? For me, it is an abstract class. I'm not a huge fan of the private constructor.

Reply:
In Druid code it's normal for classes like this (that contain only static methods) to be non-abstract and have a private constructor with a comment like "No instantiation" or "Static methods only". It's just a way of working around the fact that everything in Java must be a class, even things like this that don't really make sense as OO style classes.

Reply:
Hmm, okay. I've removed the abstract class and reverted it to the original.